ansible/post/files/approve_and_issue.sh

```diff
@@ -16,97 +16,97 @@ POWER_COUNT="${2}"
 # Var: ${self.triggers.approve}
 POWER_PREFIX="${3}"
 
-APPROVED_WORKERS=0
-ISSUED_WORKERS=0
+# Machine Prefix
+MACHINE_PREFIX="${POWER_PREFIX}-worker"
+
+# Check count for Power worker/s to be added is not zero
+if [ "0" -eq "${POWER_COUNT}" ]
+then
+  echo "Supplied Worker count is zero hence exiting."
+  exit 0
+fi
 
+# Setting values for variables
+APPROVED_WORKERS=0
 IDX=0
-while [ "$IDX" -lt "121" ]
+READY_COUNT=$(oc get nodes -l kubernetes.io/arch=ppc64le | grep "${MACHINE_PREFIX}" | grep -v NotReady | grep -c Ready)
+
+# Approce CSR and Check Ready status
+while [ "${READY_COUNT}" -ne "${POWER_COUNT}" ]
 do
+  
   export HTTPS_PROXY="http://${PROXY_SERVER}:3128"
 
-  echo "Try number: ${IDX}"
-  echo "List of Power Workers: "
+  echo "List of All Power Workers: "
   oc get nodes -l 'kubernetes.io/arch=ppc64le' -o json | jq -r '.items[] | .metadata.name'
   echo ""
 
+  echo "Approve and Issue - #${IDX}"
+  echo "List of Power Workers to be added with prefix '${POWER_PREFIX}': "
+  oc get nodes -l 'kubernetes.io/arch=ppc64le' --no-headers=true | grep "${MACHINE_PREFIX}"
+  echo ""
+
+  # Approve openshift-machine-config-operator:node-bootstrapper CSR/s
+  APPROVED_WORKERS=0
```

Comment:
```suggestion
```

Reply:
I agree APPROVED_WORKERS can be removed as it is not getting used.